### PR TITLE
Bite 933

### PR DESCRIPTION
**Pull request by AndyMoore111**
new bitesize pack for bitesize specific st2 scripts

action scripts and workflows to create, approve and provision a kubernetes namespace, add jenkins to that namespace and set a quota for it

### CODE REVIEW

File: packs/bitesize/actions/approve_ns.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        api_version.Configuration().password = password
+
+        apiclient = api_version.ApiClient(
+            master_url,
+            header_name="Authorization",
+            header_value=api_version.configuration.get_basic_auth_token())
+        apiclient.default_headers['Content-Type'] = 'application/json'
+
+        client = api_library(apiclient)
+        return client
```
Comment (pidah) [inline packs/bitesize/actions/approve_ns.py]: 1 We can abstract this class or possibly consider re-using/updating the the default class for the pack here: https://github.com/pearsontechnology/st2contrib/blob/master/packs/kubernetes/actions/lib/action.py

File: packs/bitesize/actions/install_jenkins.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        apiclient = api_version.ApiClient(
+            master_url,
+            header_name="Authorization",
+            header_value=api_version.configuration.get_basic_auth_token())
+        apiclient.default_headers['Content-Type'] = 'application/json'
+
+        client = api_library(apiclient)
+        return client
+
```
Comment (pidah) [inline packs/bitesize/actions/install_jenkins.py]: same comment as [1]

File: packs/kubernetes/actions/create_ns.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        apiclient = api_version.ApiClient(
+            master_url,
+            header_name="Authorization",
+            header_value=api_version.configuration.get_basic_auth_token())
+        apiclient.default_headers['Content-Type'] = 'application/json'
+
+        client = api_library(apiclient)
+        return client
+
```
Comment (pidah) [inline packs/kubernetes/actions/create_ns.py]: same comment as [1]

File: packs/kubernetes/actions/list_namespaces.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        apiclient = api_version.ApiClient(
+            master_url,
+            header_name="Authorization",
+            header_value=api_version.configuration.get_basic_auth_token())
+        apiclient.default_headers['Content-Type'] = 'application/json'
+
+        client = api_library(apiclient)
+        return client
+
```
Comment (pidah) [inline packs/kubernetes/actions/list_namespaces.py]: same comment as [1]

File: packs/kubernetes/actions/nslabel.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        api_version.Configuration().password = password
+
+        apiclient = api_version.ApiClient(
+            master_url,
+            header_name="Authorization",
+            header_value=api_version.configuration.get_basic_auth_token())
+        apiclient.default_headers['Content-Type'] = 'application/json'
+
+        client = api_library(apiclient)
+        return client
```
Comment (pidah) [inline packs/kubernetes/actions/nslabel.py]: same comment as [1]

File: packs/kubernetes/actions/setquota.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        api_version.Configuration().password = password
+
+        apiclient = api_version.ApiClient(
+            master_url,
+            header_name="Authorization",
+            header_value=api_version.configuration.get_basic_auth_token())
+        apiclient.default_headers['Content-Type'] = 'application/json'
+
+        client = api_library(apiclient)
+        return client
```
Comment (pidah) [inline packs/kubernetes/actions/setquota.py]: same comment as [1]

File: packs/bitesize/actions/approve_ns.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        k8suser  = self.config.get('user')
+        k8spass  = self.config.get('password')
+        k8surl   = self.config.get('kubernetes_api_url')
+
+        self.k8s = K8sClient(k8surl, k8suser, k8spass)
+
+        nsdata = self.k8s.k8s.read_namespace(ns).to_dict()
+
+        if 'metadata' in nsdata and 'labels' in nsdata['metadata'] and 'status' in nsdata['metadata']['labels']:
+            print json.dumps(self.k8s.k8s.patch_namespace(patch, ns).to_dict(), sort_keys=True, indent=2, default=self.json_serial)
```
Comment (pidah) [inline packs/bitesize/actions/approve_ns.py]: could this be `if 'status' in nsdata['metadata']['labels']:
                           print blah` ?

Comment (AndyMoore111) [inline packs/bitesize/actions/approve_ns.py]: change added to tidy this up

**Comment by pidah** [comment]
LGTM

**Comment by pidah** [comment]
LGTM

**Comment by devlinmr** [comment]
LGTM

**Comment by pidah** [comment]
LGTM

**Comment by devlinmr** [comment]
LGTM

**Pull request merged**